.github/workflows/ci-generate.yaml

```
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  pull_request:
+     branches:
+       - main
+
+jobs:
+  drift-check:
+    runs-on: ubuntu-latest
+    env:
+      REPO: ${{ github.event.repository.name }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Install cargo-generate
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-generate
+
+      - name: Retain a copy of every generated file
+        run: |
+          for template_file in $(find . -type f -name "*.template"); do
+            generated_file="${template_file%.template}"
+            if [ -f "$generated_file" ]; then
+              # Keep a copy to diff with the output of the next step
+              cp "$generated_file" "${generated_file}.check"
+            fi
+          done
+
+      - name: Generate from templates
+        run: |
+          # Note: we use cargo-generate itself here as our wrapping of the cargo-generate library
+          # does not allow us to do useful things like --allow-commands and use local templates.
+          cargo generate --path . --name "$REPO" --allow-commands
```

```suggestion
          cargo generate --path . --name '${{ env.REPO }}' --allow-commands
```
This way when you look at the workflow logs, you'll see the expanded value of `$REPO` which will make the command more readable.

---

Do you feel strongly about this one? I usually avoid echoing environment, obviously pretty harmless in this case but probably not a great pattern to establish...

---

In terms of not leaking the values to the logs? They're already exposed in the GitHub UI at the start of every step; there's a thing you can click to expand and show the entire environment.

---

> they're already exposed
I know, it's just habit I suppose. No dramas.